adafruit_esp32spi/adafruit_esp32spi_server.py

```diff
@@ -0,0 +1,193 @@
+# The MIT License (MIT)
+#
+# Copyright (c) 2019 ladyada for Adafruit Industries
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+
+"""
+`adafruit_esp32spi_server`
+================================================================================
+
+TODO: better description?
+Server management lib to make handling and responding to incoming requests much easier
+
+* Author(s): Matt Costi
+"""
+# pylint: disable=no-name-in-module
+
+import os
+from micropython import const
+import adafruit_esp32spi.adafruit_esp32spi_socket as socket
+from adafruit_esp32spi.adafruit_esp32spi_requests import parse_headers
+
+_the_interface = None   # pylint: disable=invalid-name
+def set_interface(iface):
+    """Helper to set the global internet interface"""
+    global _the_interface   # pylint: disable=global-statement, invalid-name
+    _the_interface = iface
+    socket.set_interface(iface)
+
```

Comment:
The use of a global variable always raises a flag for me, personally. However, I'm currently at a loss of how to address this situation differently. I agree that it shouldn't be instanced in the `server` class...

Anyone in @adafruit/circuitpythonlibrarians have an idea to avoid this global?

Reply:
This is same as in the other files in this library (see https://github.com/adafruit/Adafruit_CircuitPython_ESP32SPI/blob/master/adafruit_esp32spi/adafruit_esp32spi_socket.py#L37), so I think it's ok for now, we might fix it later in all the classes.

Reply:
Fair point @arturo182. I obviously didn't look to see that it was already in the lib.